### DISCUSSION TITLE
Option to Continue Generation After N Steps (IDEA)

### DISCUSSION THREAD

**Discussion by SunGreen777** [lllyasviel/stable-diffusion-webui-forge#2124] [Ideas]
Let's assume I have already generated an ancient Greek warrior in 20 steps. I like the concept and pose, but the number of steps seemed insufficient. Is it possible to implement an option to regenerate, continuing from the 20 step? This can be a very useful option.

**Comment by ostap667inbox** [comment]
If you don't have enough 20 steps, click the 'reuse seed' button to the right of the 'seed' field and generate the same image with as many steps as you want.
I doubt anyone would want to so radically rework WebUI workflow and code for your personal infrequent cases. You do realize that you see the result after the image is transformed from latent space into visible space, resources are freed and it is impossible to 'continue' here in principle, don't you?

**Reply by SunGreen777** [reply]
> 
Hi : )
That’s exactly why I asked if this could be implemented. I understand that it's possible to repeat the generation with a higher number of steps, but my suggestion is meant to save time. If you have a 4090 with 24GB, I understand, but if it's a 3060 with 12GB, you should know that generation takes about 160 seconds.
Let’s leave the question to those who know better. Every idea has value.
As for my suggestions, it's important to understand that there’s low activity here, and if no one has supported the idea yet, it's not because no one needs it, but because they haven’t seen it.
I don’t know everything, and likely neither do you. It might be possible to set checkpoints during image generation.

**Comment by jswag245** [comment]
Isn't this what upscaling/img2img could be used for? I thought it was common for others to generate at a low number of steps and resolution and use upscaling or img2img to introduce more steps/resolution for a generation they like to save time.

**Comment by abzaloff** [comment]
I don't think this is a good idea. In that case, resources should not be released after each generation, and wait for the next user action. Start a new generation or continue the existing one. This would do more harm than good.
